ee/src/main/java/org/jboss/as/ee/component/deployers/LifecycleAnnotationParsingProcessor.java

```diff
@@ -107,6 +107,9 @@ private void processLifeCycle(final EEModuleDescription eeModuleDescription, fin
             builder.setPostConstruct(methodIdentifier);
         } else if (annotationType == PRE_DESTROY_ANNOTATION) {
             builder.setPreDestroy(methodIdentifier);
+        } else if(annotationType == AROUND_CONSTRUCT_ANNOTATION){
+            String returnType = methodInfo.returnType().name().toString();
+            builder.setAroundConstruct(MethodIdentifier.getIdentifier(returnType, methodInfo.name(),InvocationContext.class.getName()));
         } else {
             builder.setAroundConstruct(methodIdentifier);
         }
```

Comment:
This 'else' block is now unused.

I think the fix should be done earlier in the method when the object referenced by "methodIdentifier" is created. At that point you have the MethodInfo, can determine the return type, and can precisely create the correct MethodIdentifier. If non-void return types are invalid for POST_CONSTRUCT_ANNOTATION, PRE_DESTROY_ANNOTATION you can also check for and reject that, and if they are not invalid then this class is incorrect for those as well.